Update copilot-instructions.md: Handling File Redirection (#35955)
dotnet/AspNetCore.Docs: .github/copilot-instructions.md

- Open Lower Bound: `<= aspnetcore-9.0`
- Specific Version: `== aspnetcore-9.0`

### Handling File Redirections

> - When a Markdown (.md) file is deleted in a PR, create a redirection entry

Reviewer: ```suggestion
- When a Markdown (.md) article file (this does not apply to includes) is deleted in a PR, create a redirection entry.
```

- Redirections ensure users following existing links aren't left with broken links
- To add a redirection:
  1. Update the `@dotnet/AspNetCore.Docs/.openpublishing.redirection.json` file
  2. Follow this format for new entries:
     ```json
     {
         "source_path": "aspnetcore/path/to/deleted-file.md",
         "redirect_url": "/aspnet/core/path/to/target-file",
         "redirect_document_id": false
     }
     ```

>   3. Use relative URLs for redirection to other files within the Microsoft Learn documentation

Reviewer: ```suggestion
  3. Use relative URLs for redirection to pages in the `learn.microsoft.com` domain
```
Note that bulleted or numbered list entries should end with a period. That rule applies if any entry in the list is a complete sentence independently or including the lead-in phrase.

     - Example: `/aspnet/core/path/to/target-file`

>   4. For external URLs, use absolute URLs including the domain

Reviewer: ```suggestion
  4. For URLs in a different domain, use absolute URLs including the domain.
```
